src/main/java/com/qaprosoft/carina/demo/gui/pages/OpinionsPage.java

```diff
@@ -0,0 +1,225 @@
+
+    @FindBy(id = "sort-comments")
+    private ExtendedWebElement sortBy;
+
+    @FindBy(xpath = "//option[@value='0']")
+    private ExtendedWebElement newestFirst;
+
+    @FindBy(xpath = "//option[@value='2']")
+    private ExtendedWebElement bestRating;
+
+    @FindBy(xpath = "//div[@class='user-thread']")
+    private List<Comment> commentsList;
+
+    public OpinionsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public void sortByBestRating() {
+        if (!bestRating.getElement().isSelected()) {
+            sortBy.click();
+            bestRating.click();
+        }
+    }
+
+    public void sortByNewestFirst() {
+        if (!newestFirst.getElement().isSelected()) {
+            sortBy.click();
+            newestFirst.click();
+        }
+    }
+
+    public boolean isPossibleSortByBestRating() {
+        if (sortBy.isElementPresent()) {
+            LOGGER.info("Sort possible.");
+        } else {
+            LOGGER.error("Sort impossible!");
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isCommentsSortByBestRating() {
+        for (int i = 0; i < commentsList.size() - 1; i++) {
+            for (int j = i + 1; j < commentsList.size(); j++) {
+                if (Integer.parseInt(commentsList.get(i).getScoreComment().getText()) >= Integer.parseInt(commentsList.get(j).getScoreComment().getText())) {
+                    LOGGER.info(commentsList.get(i).getScoreComment().getText() + " more or equals " + commentsList.get(j).getScoreComment().getText());
+                } else {
+                    LOGGER.error(commentsList.get(i).getScoreComment().getText() + " less than " + commentsList.get(j).getScoreComment().getText());
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    public void verifyGoodRateComment() {
+        int beforeGoodRate = Integer.parseInt(commentsList.get(0).getScoreComment().getText());
+        commentsList.get(0).upvoteRating();
+        int afterGoodRate = Integer.parseInt(commentsList.get(0).getScoreComment().getText());
+        Assert.assertTrue(beforeGoodRate < afterGoodRate, "Comment rating was not upvoted -> (last rating = " + beforeGoodRate + " and new rating = " + afterGoodRate + ")!");
+    }
+
+    public boolean isCommentsSortByNewestFirst() {
+        //  Constants
+        final String SPLIT_SYMBOL = " ";
+        final String MINUTES = "minutes";
+        final String HOURS = "hours";
+
+        //  First date for compare
+        LocalDate firstLocalDate;
+        //  Second date for compare
+        LocalDate secondLocalDate;
+        //  List short name moths
+        List<String> shortNameMonths = new ArrayList<>(Arrays.asList("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"));
+
+        //  Comparison two dates
+        for (int i = 0; i < commentsList.size() - 1; i++) {
+            //  Array of strings with the first comment's date
```

Comment:
Logic for parsing String to LocalDate the same for first and second date.How about create separate class like DateUtil which includes method for parsing date from String to LocalDate.I think it will reduce amount of code in this method)